### PR TITLE
Do not reassign entity ids on world changes / respawns

### PR DESCRIPTION
**Pull request by Johni0702**
This PR fixes a bug caused by when teleporting entities to other worlds due to which entities are displayed multiple times to the clients.
This is also the cause for players hanging around after death which is also fixed with this PR.
Entities near the respawn location being invisible is also fixed with this PR.
**Current behavior**
Entity ids are allocated on a per world basis in the `EntityManager` and are reassigned if entities change the world or players respawn. This leads to the old id not being removed and artifacts being created. Furthermore the entity id should not change at all (due to CraftBukkit compatibility).
**New behaviour**
Entity ids are now allocated in the `EntityIdManager` of which only a single instance exists per server. This allows the entity to change its world without being reassigned  to a new id by the `EntityManager` of the new world.
This PR also resets all known entities when a player respawns in the same world. This causes all entities to be resend after the respawn which solves the third issue.

**Related Links:**
- Ticket: #486 
- Ticket: #447

### CODE REVIEW

**Comment by turt2live** [comment]
Because we have another artifact problem I will be testing that as well. If the fix is at all related, I will expect a fix for it here :)  (death artifacts)

**Comment by Johni0702** [comment]
The death artifacts do indeed have the same reason. The only difference is that players don't get teleported to another world and therefore we have to check whether their id has changed. Previous commit fixed that.
I edited the description to also include that ticket.

**Comment by turt2live** [comment]
Due to CraftBukkit API compatibility, this is an incorrect fix. The entity ID shouldn't be changing, fixing that would be the correct fix. Please make the required adjustments, thanks!

**Comment by Johni0702** [comment]
@turt2live Changed PR content and updated title and description accordingly.

File: src/main/java/net/glowstone/entity/GlowPlayer.java [hunk 1]
```diff
@@ -629,6 +629,15 @@ public void respawn() {
         // fire event and perform spawn
         PlayerRespawnEvent event = new PlayerRespawnEvent(this, dest, spawnAtBed);
         EventFactory.callEvent(event);
+        if (event.getRespawnLocation().getWorld() == location.getWorld() && knownEntities.size() > 0) {
```
Comment (turt2live) [inline src/main/java/net/glowstone/entity/GlowPlayer.java]: Use `.equals`

**Comment by SpaceManiac** [comment]
Merged in 37f01ea44298df86cf45756e59082df4abb60c8c.

**Pull request closed without merging**